common/static/coffee/src/discussion/views/discussion_content_view.coffee

```diff
@@ -220,12 +222,14 @@ if Backbone?
       url = @model.urlFor(if is_voting then "upvote" else "unvote")
       updates =
         upvoted_ids: (if is_voting then _.union else _.difference)(user.get('upvoted_ids'), [@model.id])
-      DiscussionUtil.updateWithUndo(
+      promise = DiscussionUtil.updateWithUndo(
         user,
         updates,
         {url: url, type: "POST", $elem: $(event.currentTarget)},
         gettext("We had some trouble saving your vote.  Please try again.")
-      ).done(() => if is_voting then @model.vote() else @model.unvote())
+      )
+      if promise
+        promise.done(() => if is_voting then @model.vote() else @model.unvote())
```

Comment:
@MatMoore just a heads up, if safeAjax would always return promise as I suggested above these changes could be reverted as well.